lfs/gitscanner_log.go

```diff
@@ -279,51 +280,70 @@ func (s *logScanner) finishLastPointer() *WrappedPointer {
 // There can be multiple diffs per commit (multiple binaries)
 // Also when a binary is changed the diff will include a '-' line for the old SHA
 func (s *logScanner) scan() (*WrappedPointer, bool) {
-	for s.s.Scan() {
-		line := s.s.Text()
-
-		if match := s.commitHeaderRegex.FindStringSubmatch(line); match != nil {
-			// Currently we're not pulling out commit groupings, but could if we wanted
-			// This just acts as a delimiter for finishing a multiline pointer
-			if p := s.finishLastPointer(); p != nil {
-				return p, true
-			}
-		} else if match := s.fileHeaderRegex.FindStringSubmatch(line); match != nil {
-			// Finding a regular file header
-			p := s.finishLastPointer()
+	long := false
+	for {
+		bytes, prefix, err := s.r.ReadLine()
 
-			// Pertinent file name depends on whether we're listening to additions or removals
-			if s.dir == LogDiffAdditions {
-				s.setFilename(match[2])
+		if err != nil {
+			if err == io.EOF {
+				break
 			} else {
-				s.setFilename(match[1])
+				s.err = err
+				return nil, false
 			}
+		}
 
-			if p != nil {
-				return p, true
-			}
-		} else if match := s.fileMergeHeaderRegex.FindStringSubmatch(line); match != nil {
-			// Git merge file header is a little different, only one file
-			p := s.finishLastPointer()
+		if prefix {
+			long = true
+		} else if long {
+			long = false
+			tracerx.Printf("Skipping long line ending with: %v", string(bytes))
+		} else {
```

Comment:
Sorry for the extra comment -- while reading this over again, I realize it would also be good to change from `prefix` to `isPrefix`, to make it more clear that it's a boolean value.